### PR TITLE
Update docs for new demo servers

### PR DESCRIPTION
**Pull request by stevejalim**
## One-line summary

This changeset updates the Bedrock docs with how to use the new Demo-server infrastructure. 

## Significant changes and points to review

Currently, the new approach is not active for Mozorg demos, only Pocket demos, but we will arrange a switch over v soon, in a way that doesn't impact everyone's workflows. At that point, we can merge these docs, but pre-reading them now is useful, particularly if they raise questions.

We also add an Architecture Decision Record to capture the move from Heroku demo servers to GCP demo servers.

## Testing
To test this work:

- [ ] `make docs`
- [ ] Review http://0.0.0.0:8100/contribute.html#server-architecture

### CODE REVIEW

File: docs/contribute.rst [hunk 1]
```diff
@@ -112,40 +112,93 @@ Server architecture
...
+   - Branch ``pocket-demo-3`` -> https://www-demo3.tekcopteg.com/
+   - Branch ``pocket-demo-4`` -> https://www-demo4.tekcopteg.com/
+   - Branch ``pocket-demo-5`` -> https://www-demo5.tekcopteg.com/
+
+For example, for Mozorg:
 
 .. code-block:: bash
 
-  $ git push -f mozilla my-demo-branch:demo/1
+  $ git push -f mozilla HEAD:mozorg-demo-2
```
Comment (alexgibson) [inline docs/contribute.rst:139]: Nit: maybe this is just personal preference, but `git push -f mozilla my-demo-branch:mozorg-demo-1` feels more explicit to me in terms of an example. Possibly less prone to an accidental wrong-branch push as well?

File: docs/contribute.rst [hunk 2]
```diff
@@ -112,40 +112,93 @@ Server architecture
...
+Both Mozorg and Pocket mode have specific demo-use-only env var files, which
+are only used by our GCP demo setup. They are:
 
-.. Note::
+* ``bedrock/gcp/bedrock-demos/cloudrun/mozorg-demo.env.yaml``
+* ``bedrock/gcp/bedrock-demos/cloudrun/pocket-demo.env.yaml``
+
+If you need to set/add/remove an env var, you can edit the relevant file on
+your feature branch, commit it and push it along with the rest of
+the code, as above. There is a small risk of clashes, but these can be best
```
Comment (alexgibson) [inline docs/contribute.rst:170]: Curious, what kind of clashes might we see here?

**Review by alexgibson — APPROVED** [review]
Everything reads well and the demo push workflow seems like it will be nice and straightforward. Nice work! r+

File: docs/contribute.rst (hunk 1 above)
Comment (stevejalim) [reply docs/contribute.rst:139]: Sure thing - will amend

**Pull request merged**